src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -100,7 +100,11 @@
                         ;; Try to get a connection to make sure the db is ready
                         (.close (.getConnection datasource))
                         (when-let [{:keys [migration-db migration-dir replication-mode]} migration-opts]
-                          (when-not (= replication-mode :replica)
+                          (if (= replication-mode :replica)
```

Comment:
A comment here about the reason we do this when configured as a replica would be very helpful.